### PR TITLE
feat: implement login with magic link

### LINKED ISSUE #18: [FEAT] - Authentication Page -> Login with magic link
### Description
Implement the Login with magic link page as shown in the design

### Acceptance Criteria
- Responsive across all screen sizes (desktop, tablet and mobile)
- Write tests for your code.
- Use components where necessary (header, footer, buttons, modal and so on)

### Timeframe
24 hours

[Link to Design](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-868&t=p2gECcZQmYqvvrn7-4)

### Preview
image

### PR DESCRIPTION
**Pull request by iamExony**
# Description
Implement the Login with magic link page

**Closes #18**

# Changes proposed

## What were you told to do?
I was instructed to implement a responsive login page with a magic link feature, write tests for the code, and use components where necessary (such as the header, footer, buttons, and modal).
Figma Design:
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-868&t=mpiMe1cp8NXRwgLd-0

[FEAT] - Authentication Page -> Login with magic link 

## What did you do?

- Implementatd a magic login
- Add Email validation to check valid email address
- Tested my code using vitest with a pass
- My codes has no ESLint errors
- Complemented a responsive design on mobile, desktop and tablet screens
- Ensure the UI matches with the figma design

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x ] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
image
image
Lintest
test
testfull

### CODE REVIEW

File: src/components/common/MagicLogin/MagicLogin.tsx
Comment (Justfemi) [inline src/components/common/MagicLogin/MagicLogin.tsx:1]: Your file shouldn't be in this directory, add it to the auth folder please

**Review by Justfemi — CHANGES_REQUESTED** [review]

**Review by oreoluwa212 — APPROVED** [review]
move your folder into the appropriate folder as stated

**Comment by iamExony** [comment]
> move your folder into the appropriate folder as stated

I have placed it in app/auth. cos that the only auth i can find. if there any other thing please let me know. thank you

**Comment by JulianaSau** [comment]
your lint tests are failing

**Comment by iamExony** [comment]
> your lint tests are failing

Ma the lint test passed, the vitest failed not from my own task but someone elses task. the page.tsx from someone else mine passed correctly. mine is Login.tsx. I have made the corrections

**Review by mrcoded — APPROVED** [review]

**Review by mrcoded — CHANGES_REQUESTED** [review]
move this file to the auth folder @iamExony

**Review by mrcoded — APPROVED** [review]

**Pull request merged**